### ISSUE TITLE
(2.4.3) Soldiers attack the walls even if it is destroyed

### ISSUE THREAD

**Issue by SimoneStarace** [Wargus/wargus#334]
I was playing Wargus just to test it out and during the second mission of the human campaign I have to destroy the walls to rescue the Elven Rangers.

What I found weird is when I destroyed the walls the soldiers still attacks the walls like if it those aren't destroyed.

Here is a picture of what I had:

![picture](https://i.postimg.cc/3xz1YxdG/ezgif-com-gif-maker.gif)

**Comment by ipochto** [comment]
Hi.
Which version of stratagus/wargus do you have?
This bug was fixed here: https://github.com/Wargus/stratagus/commit/5e53670f2921341fd8887a45d3fc79e7dfdd592c
At least I thought it fixed.

**Comment by SimoneStarace** [comment]
> Hi.
> Which version of stratagus/wargus do you have?
> This bug was fixed here: [Wargus/stratagus@5e53670](https://github.com/Wargus/stratagus/commit/5e53670f2921341fd8887a45d3fc79e7dfdd592c)
> At least I thought it fixed.

Yes I forgot to write some info:

- Wargus version 2.4.3
- Played on Zorin OS Lite 15.3

Other bugs I found while I was playing were those:
- I was playing the 5th orc mission/chapter and while I was losing I decided to load a save file, which was simply a save file that loads the 4th orc mission/chapter. This is nothing strange but during the 4th chapter I have made some [Troll Destroyers](https://wowwiki.fandom.com/wiki/Destroyer) but what I have found weird was the sprite of the ships was a complete mess. It was a mix of terrain and humans walls.
- During my play on the 4th orc mission/chapter I have made transports to bring the units to the human lands. For some strange reason one ship started to move slowly and then stopped to move. Even if I was able to select it and giving orders the orc transport didn't move.
- The ogre-mage spell [Bloodlust](https://wow.gamepedia.com/Ogre-Mage_(Warcraft_II)#Spells) applied on units should have an icon on them to tell that unit has affected by the spell. What I have got is the icon wasn't the red drop but a mix of gray pixels.
